pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -170,27 +182,82 @@ func remediate(r *ReconcileMachineHealthCheck, machine *mapiv1.Machine) (reconci
 		return reconcile.Result{}, err
 	}
 
-	if !isHealthy(node) {
-		if unhealthyForTooLong(node) {
-			// delete
+	cmUnhealthConditions := &corev1.ConfigMap{}
+	cmKey := types.NamespacedName{
+		Name:      healthcheckingv1alpha1.ConfigMapNodeUnhealthConditions,
+		Namespace: r.namespace,
+	}
+	err = r.client.Get(context.TODO(), cmKey, cmUnhealthConditions)
+	if err != nil {
+		// Error reading the object - requeue the request
+		if !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+
+		// creates dummy config map with default values if it does not exist
+		cmUnhealthConditions, err = conditions.CreateDummyUnhealthyConditionsConfigMap()
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		glog.Warningf(
+			"ConfigMap %s not found under the namespace %s, fallback to default values: %s",
+			healthcheckingv1alpha1.ConfigMapNodeUnhealthConditions,
+			r.namespace,
+			cmUnhealthConditions.Data["conditions"],
+		)
+	}
+	nodeUnhealthConditions, err := conditions.GetNodeUnhealthyConditions(node, cmUnhealthConditions)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	var result *reconcile.Result
+	for _, c := range nodeUnhealthConditions {
```

Comment:
Default deployments will always have either empty CM or no CM. In the first case, the machine controller will not do anything until the CM is updated with valid data by third party controller. In the second case, the machine controller will fall into failed-all-the-time loop. Is this the expected behavior? @enxebre WDYT?

Reply:
I see, I did not see it as it was hidden in `getUnhealthConditionsConfigMap` call. Ok.